src/org/exist/Indexer.java

```diff
@@ -305,6 +305,9 @@ public void endDTD() {
     @Override
     public void endDocument() {
         if (!validate) {
+            if(indexListener != null) {
+                indexListener.endDocument(transaction);
```

Comment:
`listener.startDocument` and `listener.endDocument` will not confuse you when you read that code, right?